sycl/unittests/windows/dllmain.cpp

```diff
@@ -0,0 +1,65 @@
+//==----- dllmain.cpp --- verify behaviour of lib on process termination ---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+/*
+ * This test calls DllMain on Windows. This means, the process performs actions
+ * which are required for library unload. That said, the test requires to be a
+ * distinct binary executable.
+ */
+
+#include <CL/sycl.hpp>
+#include <helpers/CommonRedefinitions.hpp>
+#include <helpers/PiImage.hpp>
+#include <helpers/PiMock.hpp>
+
+#include <gtest/gtest.h>
+
+#ifdef _WIN32
+#include <windows.h>
+
+extern "C" BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason,
+                               LPVOID lpReserved);
+
+static std::atomic<int> TearDownCalls{0};
```

Comment:
Nit. I would add a `static_assert` to check that this atomic is trivially destructible.